src/modules/nft/types.ts

```diff
@@ -15,40 +17,90 @@ export interface CollectionResource {
   createdAt: Date;
 }
 
-export interface ItemResource {
-  // id is the NFT mint address.
-  id: string;
+export interface NFTItem {
+  // NFT mint address.
+  mintAddress: string;
   // name is the NFT name.
   name: string;
   // image is the image url of NFT.
   image: string;
   // owner is the owner of NFT.
-  owner?: ItemOwnerResource | null;
+  owner?: OwnerInfo | null;
   // collected indicates if collected by present member.
   collected?: boolean;
   // categories is inherited from collection.
   categories?: string[];
+  // metaplex metadata
+  metaplexMetadata: MetaplexMetadata;
+  // external metadata
+  externalMetadata: ExternalMetadata;
 }
 
-export interface ItemOwnerResource {
-  // id is the public address of owner.
-  id: string;
+export interface MetaplexMetadata {
+  symbol: string;
+  primarySaleHappened: boolean;
+  sellerFeeBasisPoints: integer;
+  isMutable: boolean;
+  tokenStandard?: string;
+  uses?: MetaplexMetadataNFTUses;
+  collection?: {
+    verified: boolean;
+    key: string;
+  };
+  creators: {
+    address: string;
+    verified: boolean;
+    share: number
+  }[];
+}
+
+export interface MetaplexMetadataNFTUses {
+  useMethod: string;
+  remaining: integer;
+  total: integer
+}
+
+export interface ExternalMetadata {
+  collection?: {
+    name: string;
+    family: string;
+  };
+  attributes?: {
```

Comment:
我看是回 array，是不是可以考慮拿掉 undefined case
```suggestion
  attributes: {
```

Reply:
會有 null 的 case.

Reply:
👌🏻